source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,143 @@
+you to run a series of operations that do not change any data until the
+transaction is committed. If any operation in the transaction returns an
+error, the driver cancels the transaction and discards all data changes
+before they ever become visible.
+
+In MongoDB, transactions run within logical **sessions**. A
+:manual:`session </reference/server-sessions/>` is a grouping of related
+read or write operations that you intend to run sequentially. Sessions
+enable :manual:`causal consistency
+</core/read-isolation-consistency-recency/#causal-consistency>` for a
+group of operations or allow you to execute operations in an
+:website:`ACID transaction </basics/acid-transactions>`. MongoDB
+guarantees that the data involved in your transaction operations remains
+consistent, even if the operations encounter unexpected errors.
+
+When using the {+driver-short+}, you can create a new session from a 
+``MongoClient`` instance as a ``ClientSession`` type. We recommend that you reuse
+your client for multiple sessions and transactions instead of
+instantiating a new client each time.
+
+.. warning::
+
+   Use a ``ClientSession`` only with the ``MongoClient`` (or associated
+   ``MongoDatabase`` or ``MongoCollection``) that created it. Using a
+   ``ClientSession`` with a different ``MongoClient`` results in operation
+   errors.
+
+Methods
+-------
+
+Create a ``ClientSession`` by using the ``startSession()`` method on your 
+``MongoClient``instance. You can then modify the session state by using the 
+methods provided by the ``ClientSession``. The following table details the 
+methods you can use to manage your transaction:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction for this session with the
+         default transaction options. Use the ``TransactionOptions`` 
+         parameter to start a transaction with given options. You 
+         cannot start a transaction if there's already an active transaction
+         on the session.
+       |
+       | **Parameter**: ``transactionOptions``
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. Returns an error 
+         if there is no active transaction for the
+         session or the transaction was previously ended.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or if the
+         transaction was ended.
+
+   * - ``withTransaction()``
+     - | Starts a new transaction for this session and runs the given function.
+       |
+       | **Parameter**: ``transactionBody``
+
+A ``ClientSession`` also has methods to retrieve session properties and modify mutable 
+session properties. View the `API documentation <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/ClientSession.html>`__ 
+to learn more about these methods.
+
```

Comment:
S: instead of linking to clientsession api docs, just reroute to the api docs section at the bottom of this guide

Reply:
QQ what's the syntax for this in rst? I didn't see this in the slide deck and what I found in other references didn't seem to work. Still getting used to using this instead of markdown!

Reply:
Think I figured it out actually!